packages/grafana-runtime/src/config.ts

```diff
@@ -306,11 +306,19 @@ function overrideFeatureTogglesFromUrl(config: GrafanaBootConfig) {
   });
 }
 
-const bootData = (window as any).grafanaBootData || {
-  settings: {},
-  user: {},
-  navTree: [],
-};
+let bootData = (window as any).grafanaBootData;
+
+if (!bootData) {
+  if (process.env.NODE_ENV !== 'test') {
+    console.error('window.grafanaBootData was not set by the time config was initialized');
+  }
+
+  bootData = {
+    settings: {},
+    user: {},
+    navTree: [],
+  };
+}
```

Comment:
Does this fallback only really exist for test purposes then? Do we expect any scenario where it actually could/should happen in production?

Reply:
But actually, not even. It's mocked in tests in jest-setup.js

https://github.com/grafana/grafana/blob/ee8205f12a08eb764061110f96e8dfdb2ad82a32/public/test/jest-setup.ts#L20-L31